tests/test_mpirun.py

```diff
@@ -10,17 +10,19 @@
 
 
 @pytest.mark.skipif(sys.platform.startswith("darwin"), reason="skipping macOS test as problem with file in pytest")
+@pytest.mark.parametrize('pset_mode', ['soa', 'aos'])
 @pytest.mark.parametrize('repeatdt', [200*86400, 10*86400])
 @pytest.mark.parametrize('maxage', [600*86400, 10*86400])
-def test_mpi_run(tmpdir, repeatdt, maxage):
+def test_mpi_run(pset_mode, tmpdir, repeatdt, maxage):
     if MPI:
         stommel_file = path.join(path.dirname(__file__), '..', 'parcels',
                                  'examples', 'example_stommel.py')
         outputMPI = tmpdir.join('StommelMPI.nc')
         outputNoMPI = tmpdir.join('StommelNoMPI.nc')
 
-        system('mpirun -np 2 python %s -p 4 -o %s -r %d -a %d' % (stommel_file, outputMPI, repeatdt, maxage))
-        system('python %s -p 4 -o %s -r %d -a %d' % (stommel_file, outputNoMPI, repeatdt, maxage))
+        # ehm, this test doesn't work. I remember that we only employ the actual ParticleSet split _when_ # particles >= 1024. Testing it with 4 particles doesn't test MPI at all.
+        system('mpirun -np 2 python %s -p 4 -o %s -r %d -a %d -psm %s' % (stommel_file, outputMPI, repeatdt, maxage, pset_mode))
+        system('python %s -p 4 -o %s -r %d -a %d -psm %s' % (stommel_file, outputNoMPI, repeatdt, maxage, pset_mode))
```

Comment:
@erikvansebille this part of the test really needs to change cause it doesn't really test MPI ...

Reply:
of course, drawback is that the test will take longer. I just don't know why we would need to test such long runtimes. Perhaps you can explain the rationale tomorrow.

Reply:
I checked for your thought that MPI is only used when p>1024, but can't find that in the code at https://github.com/OceanParcels/parcels/blob/master/parcels/collection/collectionsoa.py#L92-L119. So I think it does actually check MPI already

Note that the intention here was to test the _output_ of MPI runs; to make sure that the merging of temporary files from different nodes worked well. It does not test the efficiency etc of MPI; we will need to develop other tests for that

Reply:
so, indeed, we probably got rid of that 1024-min-particleset-size limit when redesigning the collections - which is good.

On your note: yes, sure, we test if it works, not if it's speedy. Still, if the limit was in place, the code would not have done the actual MPI stuff that the test is supposed to 'test', so with the limit the test would be meaningless.

Luckily, the artificial size limit isn't active anymore, so the test works as originally intended. Thanks for looking it up!